pkgs/applications/window-managers/dwl/default.nix

```diff
@@ -1,36 +1,44 @@
 { lib
 , stdenv
 , fetchFromGitHub
+, installShellFiles
+, libX11
 , libinput
 , libxcb
 , libxkbcommon
 , pixman
 , pkg-config
+, substituteAll
 , wayland
 , wayland-protocols
-, wlroots
+, wlroots_0_16
 , writeText
-, enable-xwayland ? true, xwayland, libX11
+, xcbutilwm
+, xwayland
+, enableXWayland ? true
 , conf ? null
 , patches ? [ ]
 }:
 
 let
   totalPatches = patches ++ [ ];
+  wlroots = wlroots_0_16;
 in
-
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
```

Comment:
```suggestion
stdenv.mkDerivation (final: {
```

Reply:
Cosmetical, and I don't like the `final` word alone, isolated. final-what?

Reply:
My thinking was it would be a bit more like the `final: prev:` overlay pattern, which was comforting. rec -> fix is already pretty cosmetical.

Reply:
Makes sense, I didn't think about this before.

However I don´t like these names for them, self-super was better.

Reply:
`self:` is fine by me

Reply:
I liked it!